intel-sgx/pcs/src/qe_identity.rs

```diff
@@ -352,13 +366,24 @@ impl QeIdentitySigned {
             isvprodid,
             tcb_levels,
             type_: PhantomData,
-        })
+        };
+
+        let now = Utc::now();
+        if now < identity.issue_date {
+            return Err(Error::Qe3IdNotValid(format!("QE3 only valid from {}", identity.issue_date)))
+        }
+        if identity.next_update < now {
+            return Err(Error::Qe3IdNotValid(format!("QE3 expired on {}", identity.next_update)))
+        }
```

Comment:
This seems like a weird location for this check. Why don't you do this check prior to constructing the `Verified` struct, like the version check?